doc/topics/the-event-loop-timers-and-nexttick

```diff
@@ -0,0 +1,174 @@
+implementation, but that's not important for this demonstration.  The most
+important parts are here.  There are actually seven or eight steps, but the
+ones we care about--ones that Node actually uses are these four.*
+
+## timers
+
+This phase executes callbacks scheduled by `setTimeout()` and `setInterval()`.
+When you create a timer, you make a call to setTimeout().  The event loop will
+eventually enter the 'poll' phase which determines how many milliseconds remain
+until the next timer. If there is a timer, it will wait for connections for that
+many milliseconds. After that many milliseconds, it will break the 'poll' phase
+and wrap back around to the timers phase where those callbacks can be processed.
+
+*Note: The 'poll' phase technically controls when timers are called due to its
+ability to cause a thread to sit idly without burning CPU in order to stall the
+event loop so the timer can execute.*
+
+## pending callbacks:
+
+This phase executes callbacks for specific types of TCP errors, for example.
+
+## poll:
+
+This is the phase in which the event loop sits and waits for incoming
+connections to be received.  Ideally, most scripts spend most of their time here.
+
+## setImmediate:
+
+`process.setImmediate()` is actually a special timer that runs in a separate
+phase of the event loop.  It uses a libuv API that schedules callbacks to execute
+after the poll phase has completed.
+
+Generally, as the code is executed, the event loop will eventually hit the
+'poll' phase where it will wait for an incoming connection, request, etc.
+However, after a callback has been scheduled with `setImmediate()`, at the start
+of the poll phase, a check will be run to see if there are any callbacks
+waiting.  If there are none waiting, the poll phase will end and continue to the
+`setImmediate` callback phase.
+
+### setImmediate vs setTimeout
+
+How quickly a `setImmediate()` callback is executed is only limited by how
+quickly the event loop can be processed whereas a timer won't fire until the
+number of milliseconds passed have elapsed.
+
+The advantage to using `setImmediate()` over `setTimeout()` is that the lowest
+value you may set a timer's delay to is 1 ms (0 is coerced to 1), which doesn't
+seem like much time to us humans, but it's actually pretty slow compared to how
+quickly `setImmediate()` can execute--the event loop operates on the microsecond
+scale (1 ms = 1000 µs).
+
+## nextTick:
+
+### Understanding nextTick()
+
+You may have noticed that `nextTick()` was not displayed in the diagram, even
+though its a part of the asynchronous API.  This is because nextTick is not
+technically part of the event loop.  Instead, it is executed at the end of each
+phase of the event loop.
+
+Looking back at our diagram, any time you call nextTick in a given phase, all
+callbacks passed to nextTick will be resolved before the event loop continues.
+This can create some bad situations because **it allows you to asynchronously
```

Comment:
I might suggest nixing `asynchronously` here, since it's synchronously blocking asynchronous events.